flow/project.py

```diff
@@ -405,11 +414,13 @@ def __init__(self, cmd, pre=None, post=None, directives=None):
     def __str__(self):
         return "{type}(cmd='{cmd}')".format(type=type(self).__name__, cmd=self._cmd)
 
-    def eligible(self, job):
+    def eligible(self, job, ignore_conditions=IgnoreConditions.NONE):
         "Eligible, when all pre-conditions are true and at least one post-condition is false."
-        pre = all(cond(job) for cond in self._prereqs)
+        pre = (ignore_conditions & IgnoreConditions.PRE) \
```

Comment:
For `pre` we don't do the length check like we do with post. I think consistency would be nice. In addition, since we are dealing with booleans, we can just use multiple conditions like `bool = cond1 or cond2 or cond3` since `or` short circuits.